### ISSUE TITLE
WSL: Training on CPU even when having compatible NVIDIA GPU

### ISSUE THREAD

**Issue by Elameri** [ivadomed/ivadomed#953]
## Issue description
Ivadomed doing training on cpu even though i have a compatible nvidia gpu and installed via ```requirements_gpu.txt```

### Current behavior
Training on cpu

### Expected behavior
Training on gpu

### Steps to reproduce
All steps in https://ivadomed.org/en/latest/installation.html with approach 2

## Environment

Windows 10 Pro : (WSL1 Ubuntu 20.04 with python 3.7) and (WSL2 Ubuntu 18.04 with python 3.8)
Nvidia Geforce gtx 1050

## Solutions tried
Tried https://docs.nvidia.com/cuda/wsl-user-guide/index.html, but the current insider preview build (which is a version of Windows 11) is not stable (crashing at every startup), had to do a clean installation of Windows 10 afterwards.

**Comment by jcohenadad** [comment]
@Elameri could you please add a complete output of your Terminal-- it might contain some useful info for debugging

**Comment by dyt811** [comment]
Hey, @Elameri. Just a question, if you have Windows OS, and IvadoMed support Windows OS natively, why do you still need to WSL again? You should be able to just setup IvadoMed on Windows with Python/Pip/Conda/Torch etc and get the training going and should be able to run without issues?

**Comment by dyt811** [comment]
The chance of you succeeding in Windows natively without WSL is much higher. You might want to check `yw/python3.9_roadblocks` branch as we have been testing that a bit more with more modern version of Torch (1.8) and you might less likely be running into compatibility issues with CUDA. 

Batch/CMD/Powershell, all of them are okay. Before trying too far, validate via

```
import torch
torch.cuda.is_available()
```

To ensure Cuda/Torch is properly setup before trying anything IvadoMed related.

**Comment by jcohenadad** [comment]
> Hey @dyt811, I didn't know ivadomed supports windows natively because I read in https://ivadomed.org/en/latest/installation.html "Currently, we only support MacOS and Linux operating systems. Windows users have the possibility to install and use ivadomed via Windows Subsystem for Linux (WSL)".

@dyt811 if windows is supported natively, this is something we need to fix in the doc

**Comment by dyt811** [comment]
Yup. Will update this ASAP. Windows Torch CPU based IvadoMed has been running and well tested through GitHub Action CI processes for a long time post PR #757 but the Windows GPU based approaches has not undergone the same systematic tests. Going to test it out shortly before the documentation update.

**Comment by dyt811** [comment]
Also, kudos to @kousu who did some preliminary testing/digging with GitHub Actions runner to run on romane at experimental #955. We both feel a bit uneasy to potentially expose GPUs for public use (i.e. malicious PRs, as [reported earlier](https://neuropoly.slack.com/archives/C034UD4QW/p1617755387075200) from a public repo). We are exploring potentially more private options at the moment via small proof of concept setups.

**Comment by jcohenadad** [comment]
> We both feel a bit uneasy to potentially expose GPUs for public use (i.e. malicious PRs, as reported earlier from a public repo). We are exploring potentially more private options at the moment via small proof of concept setups.

what is the current GPU offer by GitHub actions, google colab, etc. i'm sure we are not the only one with this need. How do others do?

**Comment by kousu** [comment]
> > We both feel a bit uneasy to potentially expose GPUs for public use (i.e. malicious PRs, as reported earlier from a public repo). We are exploring potentially more private options at the moment via small proof of concept setups.
> 
> what is the current GPU offer by GitHub actions, google colab, etc. i'm sure we are not the only one with this need. How do others do?

Move this to https://github.com/neuropoly/computers/issues/202
